vllm/model_executor/models/mllama4.py

```diff
@@ -818,9 +914,12 @@ def load_weights(self, weights: Iterable[tuple[str,
         assert loaded_language_model_params is not None
         updated_params.update(loaded_language_model_params)
 
+        if self.use_data_parallel:
+            other_weights = self._consolidate_qkv_weights(other_weights)
```

Comment:
nit: we can add some sanity checks to ensure _consolidate_qkv_weights operate appropriately.